swift/Samples/Tutorial/Tutorial2.md

```diff
@@ -0,0 +1,597 @@
+
+Anytime a child workflow is run, the parent needs a way of converting its `Output` into an `Action` it can handle. The `WelcomeWorkflow`'s output type is currently an empty enum: `enum Output { }`. For now, delete the `Output` on `WelcomeWorkflow` and replace it with a typealias to `Never`:
+
+```swift
+struct WelcomeWorkflow: Workflow {
+    typealias Output = Never
+}
+```
+
+Now, update the `TutorialContainerViewController` to start at the `RootWorkflow` and we'll see the welcome screen again.
+
+```swift
+public final class TutorialContainerViewController: UIViewController {
+    let containerViewController: UIViewController
+
+    public init() {
+        // ...
+
+        // Create a `ContainerViewController` with the `RootWorkflow` as the root workflow, with the view registry we just created.
+        containerViewController = ContainerViewController(
+            workflow: RootWorkflow(),
+            viewRegistry: viewRegistry)
+
+        super.init(nibName: nil, bundle: nil)
+    }
+    // ...
+}
+```
+
+### Navigating between Workflows
+
+Now that there is a root workflow, it can be updated to navigate between the `Welcome` and `TodoList` workflows.
+
+Start by defining the state that needs to be tracked at the root - specifically which screen we're showing, and the actions to login and logout:
+
+```swift
+// MARK: State and Initialization
+
+extension RootWorkflow {
+
+    // The state is an enum, and can either be on the welcome screen or the todo list.
+    // When on the todo list, it also includes the name provided on the welcome screen
+    enum State {
+        // The welcome screen via the welcome workflow will be shown
+        case welcome
+        // The todo list screen via the todo list workflow will be shown. The name will be provided to the todo list.
+        case todo(name: String)
+    }
+
+    func makeInitialState() -> RootWorkflow.State {
+        return .welcome
+    }
+    // ...
+}
+```
+
+```swift
+// MARK: Actions
+
+extension RootWorkflow {
+
+    enum Action: WorkflowAction {
+
+        typealias WorkflowType = RootWorkflow
+
+        case login(name: String)
+        case logout
+
+        func apply(toState state: inout RootWorkflow.State) -> RootWorkflow.Output? {
+
+            switch self {
+            case .login(name: let name):
+                // When the `login` action is received, change the state to `todo`.
+                state = .todo(name: name)
+            case .logout:
+                // Return to the welcome state on logout.
+                state = .welcome
+            }
+            return nil
+
+        }
+    }
+}
+```
+
+The root workflow is now modeling our states and actions. Soon we will be able to navigate between the welcome and todo list screens.
+
+### Workflow Output
+
+Workflows can only communicate with each other through then "inputs" and "outputs". When a child workflow emits an output, the parent workflow will receive it and turn it into an action they can handle.
```

Comment:
s/through then/through their/.

Time to start using "props" or "properties" instead of inputs? Either way, I think we should be describing in more detail just what inputs means here.

Reply:
Replaced usage of "inputs" with "properties" throughout the tutorials. I originally wrote these when "input" was the preferred parlance, thanks for catching.